lib/web_ui/lib/src/engine/canvaskit/image_web_codecs.dart

```diff
@@ -46,12 +66,99 @@ class CkBrowserImageDecoder implements ui.Codec {
       );
     }
 
+    final CkBrowserImageDecoder decoder = CkBrowserImageDecoder._(
+      contentType: contentType,
+      targetWidth: targetWidth,
+      targetHeight: targetHeight,
+      data: data,
+      debugSource: debugSource,
+    );
+
+    // Call once to initialize the decoder and populate late fields.
+    await decoder._getOrCreateWebDecoder();
+    return decoder;
+  }
+
+  CkBrowserImageDecoder._({
+    required this.contentType,
+    required this.targetWidth,
+    required this.targetHeight,
+    required this.data,
+    required this.debugSource,
+  });
+
+  final String contentType;
+  final int? targetWidth;
+  final int? targetHeight;
+  final Uint8List data;
+  final String debugSource;
+
+  @override
+  late int frameCount;
+
+  @override
+  late int repetitionCount;
+
+  /// Whether this decoded has been disposed of.
```

Comment:
Typo: `/// Whether this decoder has been disposed of.`

Reply:
Done.